File: Python/1149.py
n = int(input())

result = ['('] * (n - 1)
for i in range(1, n + 1):
    result.append('sin(1')
    for j in range(2, i + 1):
        result.append('-' if (j % 2 == 0) else '+')
        result.append(f'sin({j}')
    result.append(')' * i)
    result.append(f'+{n + 1 - i}')
    result.append(')')
result.pop()

s = ''.join(result)
print(s)

File: C++/1149.cpp
#include <cstddef>
#include <iostream>
#include <string>

int main() {
  std::size_t size;
  std::cin >> size;

  std::string result;
  result.append(size - 1, '(');
  for (std::size_t i = 1; i <= size; ++i) {
    result += "sin(1";
    for (std::size_t j = 2; j <= i; ++j) {
      result += (j % 2 != 0) ? '+' : '-';
      result += "sin(";
      result += std::to_string(j);
    }
    result.append(i, ')');
    result += '+';
    result += std::to_string(size + 1 - i);
    result += ')';
  }
  result.pop_back();

  std::cout << result << '\n';
  return 0;
}
